File: frontend/src/api/period.ts
import { api } from './auth'

export interface PeriodCloseResult {
  period_end: string
  locked_before: string
  closing_lines_written: number
  net_profit: string
}

export async function closePeriod(period_end: string): Promise<PeriodCloseResult> {
  const res = await api.post<PeriodCloseResult>('/period/close', { period_end })
  return res.data
}

File: backend/app/routers/period.py
from datetime import date

from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlalchemy.orm import Session

from app.auth import CompanyAdmin
from app.database import get_db
from app.services.period_close import close_period

router = APIRouter(prefix="/period", tags=["period"])


class PeriodCloseIn(BaseModel):
    period_end: date


class PeriodCloseResult(BaseModel):
    period_end: str
    locked_before: str
    closing_lines_written: int
    net_profit: str


@router.post("/close", response_model=PeriodCloseResult)
def close_period_endpoint(body: PeriodCloseIn, ctx: CompanyAdmin, db: Session = Depends(get_db)):
    _, company_id, _ = ctx
    return close_period(db, company_id, body.period_end)
